### PR TITLE
Bw/initial api arch

### PR DESCRIPTION
**Pull request by muselesscreator**
Adds initial react-router api architecture, data loaders, and selectors.

### CODE REVIEW

File: src/index.jsx [hunk 1]
```diff
@@ -1,21 +1,29 @@
...
 import ReactDOM from 'react-dom';
-import RouterRoot from './routes';
+import App from './App';
 
 import messages from './i18n';
 
 import './index.scss';
 
+const queryClient = new QueryClient({
+  queries: { retry: false },
```
Comment (adamstankiewicz) [inline src/index.jsx:18]: [curious] Curious for your thoughts on the default config settings for query retries :)

File: package.json [hunk 2]
```diff
@@ -38,6 +38,7 @@
     "@edx/frontend-component-header": "4.2.3",
     "@edx/frontend-platform": "4.5.1",
     "@edx/paragon": "^20.20.0",
+    "@edx/react-unit-test-utils": "npm:@edx/react-unit-test-utils@1.5.3",
```
Comment (adamstankiewicz) [inline package.json:41]: nit: could this be:

```json
{
  "@edx/react-unit-test-utils": "1.5.3"
}

File: src/data/services/lms/hooks/api.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+import { camelCaseObject } from '@edx/frontend-platform';
+
+import { queryKeys } from '../constants';
+import { loadORAConfigData, loadSubmissionData } from '../dataLoaders';
+import fakeData from '../fakeData';
+
+/**
+ *  A react-query data object
+ *  @typedef {Object} ReactQueryData
```
Comment (adamstankiewicz) [inline src/data/services/lms/hooks/api.js:11]: [curious] I wonder if `@tanstack/react-query` exports any existing types for `useQuery`?

Comment (muselesscreator) [reply src/data/services/lms/hooks/api.js:11]: was curious about that as well, but did not see something immediately.  Not sure the best way to import jsdoc types...

**Review by leangseu-edx — APPROVED** [review]

File: src/data/services/lms/hooks/api.js (hunk 3 above)
Comment (adamstankiewicz) [reply src/data/services/lms/hooks/api.js:11]: If `@tanstack/react-query` exports its types, you should be able to import it like any other module:

```js
import { UseQueryResult } from '@tanstack/react-query';
```
It looks like there is are types defined for its hooks: https://github.com/TkDodo/react-query/blob/c1ae82ba188fd5abda5e256cac070145e5941447/src/react/useQuery.ts#L16

The types for the hooks do appear to be exported: https://github.com/TkDodo/react-query/blob/c1ae82ba188fd5abda5e256cac070145e5941447/src/react/index.ts#L19

Only question if if JSDoc would accept a TS-defined type 😄

**Pull request merged**